src/Npgsql/NpgsqlReadBuffer.cs

```diff
@@ -444,6 +449,11 @@ internal string ReadNullTerminatedString(Encoding encoding)
 
         internal void Clear()
         {
+            if (Connector?.Connection != null && Connector.Connection.EnableStatistics)
```

Comment:
nit: Consider null propagation to merge the condition (e.g. `Connector?.Connection?.EnableStatistics ?? false`).